src/reducers/entities.js

```diff
@@ -13,24 +13,24 @@ const reducer = handleActions({
     [action.payload.entityType]: {
       ...state[action.payload.entityType],
       [action.payload.data.id]: {
-        ...get(state, `[${action.payload.entityType}][${action.payload.data.id}]`, {}),
+        ...get(Object.assign({}, state), `[${action.payload.entityType}][${action.payload.data.id}]`, {}),
         ...action.payload.data,
       },
     },
   }),
   [receiveEntities]: (state, action) => {
-    const result = { ...state }
+    const result = { ...(Object.assign({}, state)) }
     keys(action.payload.entities).forEach((entityType) => {
       // preserve old entities
-      const entitiesState = get(state, `${entityType}`, {})
+      const entitiesState = { ...(get(state, `${entityType}`, {})) }
```

Comment:
Why you need ```... ``` before the ```(get...)```?

Reply:
`get` returns a reference, this needs to be spread I think:
![image](https://user-images.githubusercontent.com/969493/35512021-8d423216-04fe-11e8-8a09-0b99ee15f866.png)

Reply:
Normally I am used to see, new data coming actions, and update something in the store at once using ```...``` or ```spread operator``` in this case it seems you need old data from the state as well, I suggest follow the same pattern. Fetch/query data (using lodash), and then at the end build the new state using: ```...oldState, dataFromAction, ...dataFetchedFromState```

Do you understand what I mean?